src/optimagic/optimization/multistart.py

```diff
@@ -12,20 +12,388 @@
+
+MultistartSamplingMethod = Literal["sobol", "random", "halton", "latin_hypercube"]
+
+
+@dataclass
+class MultistartOptions:
+    """Multistart options in optimization problems.
+
+    Attributes:
+        n_samples: The number of points at which the objective function is evaluated
+            during the exploration phase. If None, n_samples is set to 100 times the
+            number of parameters.
+        n_optimizations: The number of local optimizations to run. Defaults to 10% of
+            n_samples.
+        sampling_distribution: The distribution from which the exploration sample is
+            drawn. Allowed are "uniform" and "triangular". Defaults to "uniform".
+        sampling_method: The method used to draw the exploration sample. Allowed are
+            "sobol", "random", "halton", and "latin_hypercube". Defaults to "random".
+        sample: A sequence of PyTrees or None. If None, a sample is drawn from the
+            sampling distribution.
+        mixing_weight_method: The method used to determine the mixing weight, i,e, how
+            start parameters for local optimizations are calculated. Allowed are
+            "tiktak" and "linear", or a custom callable. Defaults to "tiktak".
+        mixing_weight_bounds: The lower and upper bounds for the mixing weight.
+            Defaults to (0.1, 0.995).
+        convergence_max_discoveries: The maximum number of discoveries for convergence.
+            Determines after how many re-descoveries of the currently best local
+            optima the multistart algorithm stops. Defaults to 2.
+        convergence_relative_params_tolerance: The relative tolerance in parameters
+            for convergence. Determines the maximum relative distance two parameter
+            vecctors can have to be considered equal. Defaults to 0.01.
+        n_cores: The number of cores to use for parallelization. Defaults to 1.
+        batch_evaluator: The evaluator to use for batch evaluation. Allowed are "joblib"
+            and "pathos", or a custom callable.
+        batch_size: The batch size for batch evaluation. Must be larger than n_cores
+            or None.
+        seed: The seed for the random number generator.
+        exploration_error_handling: The error handling for exploration errors. Allowed
+            are "raise" and "continue".
+        optimization_error_handling: The error handling for optimization errors. Allowed
+            are "raise" and "continue".
+
+    Raises:
+        InvalidMultistartError: If the multistart options cannot be processed, e.g.
+            because they do not have the correct type.
+
+    """
+
+    n_samples: int | None = None
+    n_optimizations: int | None = None
+    sampling_distribution: Literal["uniform", "triangular"] = "uniform"
+    sampling_method: MultistartSamplingMethod = "random"
+    sample: Sequence[PyTree] | None = None
+    mixing_weight_method: Literal["tiktak", "linear"] = "tiktak"
```

Comment:
Since this is re-used in the TypedDict below, the literal type should maybe be assigned to a variable.